metaflow/plugins/cards/card_client.py

```diff
@@ -0,0 +1,191 @@
+        self._html = self._card_ds.get_card_html(self.path)
+        return self._html
+
+    @property
+    def path(self):
+        return self._path
+
+    def browser(self):
+        import webbrowser
+
+        self._temp_file = tempfile.NamedTemporaryFile(suffix=".html")
+        html = self.html
+        self._temp_file.write(html.encode())
+        self._temp_file.seek(0)
+        url = "file://" + os.path.abspath(self._temp_file.name)
+        webbrowser.open(url)
+
+    def nb(self):
+        from IPython.core.display import HTML, display
+
+        display(HTML(self.html))
+
+
+class CardIterator:
+    def __init__(self, card_paths, card_ds, from_resumed=False, origin_pathspec=None):
+        self._card_paths = card_paths
+        self._card_ds = card_ds
+        self._current = 0
+        self._high = len(card_paths)
+        self.from_resumed = from_resumed
+        self.origin_pathspec = origin_pathspec
+
+    def __len__(self):
+        return self._high
+
+    def __iter__(self):
+        for idx in range(self._high):
+            yield self._get_card(idx)
+
+    def __getitem__(self, index):
+        return self._get_card(index)
+
+    def _get_card(self, index):
+        if index >= self._high:
+            raise IndexError
+        path = self._card_paths[index]
+        card_info = self._card_ds.card_info_from_path(path)
+        # todo : find card creation date and put it in client.
+        return Card(
+            self._card_ds,
+            card_info.type,
+            path,
+            html=None,
+            created_on=None,
+        )
+
+    def _make_heading(self, type):
+        return "<h1>Displaying Card Of Type : %s</h1>" % type.title()
+
+    def _wrap_html(self, html):
+        return (
+            """
+        <html>
+            <head></head>
+            <body>
+                %s
+            </body>
+        </html>
+        """
+            % html
+        )
+
+    def nb(self):
+        from IPython.core.display import HTML
+        from IPython.display import display_html
+
+        main_html = []
+        for idx, _ in enumerate(self._card_paths):
+            card = self._get_card(idx)
+            main_html.append(HTML(data=self._make_heading(card.type)))
+            main_html.append(HTML(data=card.html))
+        display_html(*main_html)
+
+    def _repr_html_(self):
+        from IPython.core.display import HTML
+        from IPython.display import display_html
+
+        main_html = []
+        for idx, _ in enumerate(self._card_paths):
+            card = self._get_card(idx)
+            main_html.append(self._make_heading(card.type))
+            main_html.append(card.html)
+        # return self._wrap_html()
+        return "\n".join(main_html)
+
+    def __next__(self):
+        self._current += 1
+        if self._current >= self._high:
+            raise StopIteration
+        return self._get_card(self._current)
+
+
+def get_cards(task, type=None, follow_resumed=True):
```

Comment:
as a matter of convenience, we could allow `task` to be a pathspec string optionally, so instead of `get_cards(Task('foo'))`, you can just do `get_cards('foo')`

Reply:
I have added this too.